src/Tempest/Database/src/Mappers/QueryToModelMapper.php

```diff
@@ -51,18 +51,40 @@ private function parse(ClassReflector $class, DatabaseModel $model, array $row):
 
             $count = count($keyParts);
 
+            // TODO: clean up and document
```

Comment:
Do et. 🙃

Reply:
I will before merging!

Reply:
Ok I won't before merging 🫣

I don't really like this part, and I need to think some more about how to improve it. I made an issue for it though: #507